### PR TITLE
feat: Mobile release process

### PR DESCRIPTION
**Pull request by kahest**
## DESCRIBE YOUR PR
*Tell us what you're changing and why. If your PR **resolves an issue**, please link it so it closes automatically.*

This PR adds a new "Release Channels" top level page that explains how the Mobile Release process works, specifically what Pre-Release and Stable mean. Based on [Internal doc](https://www.notion.so/sentry/New-Mobile-SDK-release-process-1138b10e4b5d80e0960bc8ce33da0aa9)

## IS YOUR CHANGE URGENT?  

Help us prioritize incoming PRs by letting us know when the change needs to go live.
- [ ] Urgent deadline (GA date, etc.): 
- [x] Other deadline: this week
- [ ] None: Not urgent, can wait up to 1 week+

## SLA

- Teamwork makes the dream work, so please add a reviewer to your PRs.
- Please give the docs team up to 1 week to review your PR unless you've added an urgent due date to it.
Thanks in advance for your help!

## PRE-MERGE CHECKLIST

*Make sure you've checked the following before merging your changes:*

- [x] Checked Vercel preview for correctness, including links
- [x] PR was reviewed and approved by any necessary SMEs (subject matter experts)
- [x] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## EXTRA RESOURCES

- [Sentry Docs contributor guide](https://docs.sentry.io/contributing/)

### CODE REVIEW

File: docs/platforms/android/releases/index.mdx [hunk 1]
```diff
@@ -0,0 +1,41 @@
+---
+title: Release Channels
+description: "Learn about the SDK's release channels and choose the right one."
+sidebar_order: 7000
+---
+
+This SDK uses the following ways to release SDK updates:
```
Comment (lizokm) [inline docs/platforms/android/releases/index.mdx:7]:
```suggestion
The Android SDK has the following release statuses for SDK updates:
```
It's good for search engine optimization to make this Android-specific

File: docs/platforms/android/releases/index.mdx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+## `Pre-release`
+
+We create pre-releases (alpha, beta, RC,…) for larger and potentially more impactful changes, such as new features or major versions.
```
Comment (lizokm) [inline docs/platforms/android/releases/index.mdx:11]:
```suggestion
Pre-releases (alpha, beta, RC,…) are for larger and potentially more impactful changes, such as new features or major versions.
```

File: docs/platforms/android/releases/index.mdx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+## `Latest`
+
+We continuously release major/minor/hotfix versions from the `main` branch. These releases go through all our internal quality gates and are very safe to use and intended to be the default for most teams.
```
Comment (lizokm) [inline docs/platforms/android/releases/index.mdx:15]:
```suggestion
We continuously release major/minor/hotfix versions from the `main` branch. These releases go through all our internal quality gates and are very safe to use. They're intended to be the default for most teams.
```

File: docs/platforms/android/releases/index.mdx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+## `Stable`
+
+We promote releases from `Latest` when they have been used in the field for some time and in scale, considering time since release, adoption, and other quality and stability metrics. These releases will be indicated on the releases page (https://github.com/getsentry/sentry-cocoa/releases/) with the `Stable` suffix.
```
Comment (lizokm) [inline docs/platforms/android/releases/index.mdx:19]:
```suggestion
Releases are promoted from `Latest` to `Stable` when they've been used in the field for some time and at some scale. The length of time a release has been out, the level of adoption, and other quality and stability metrics are considered. These releases will be indicated on the releases page (https://github.com/getsentry/sentry-cocoa/releases/) with the `Stable` label.
```

File: docs/platforms/android/releases/index.mdx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+## FAQ
+
+### Where can I find the latest stable release for any given SDK?
+
+On the releases pages of the SDK's GitHub repo. Stable releases will be indicated with the suffix `(Stable)` in the name of the release. You can also use the search box in the top right, and you will see the newest matching releases.
```
Comment (lizokm) [inline docs/platforms/android/releases/index.mdx:25]:
```suggestion
On the releases pages of the SDK's GitHub repo. Stable releases will be indicated with the label `(Stable)` in the name of the release. You can also use the search box in the top right to search for the newest releases.
```

File: docs/platforms/android/releases/index.mdx [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+### Does this mean our other releases are not stable?
```
Comment (lizokm) [inline docs/platforms/android/releases/index.mdx:27]:
```suggestion
### Does this mean the other releases we're using aren't stable?
```

File: docs/platforms/android/releases/index.mdx [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+No. Stability and performance are the top priorities for every release. The `Stable` release channel is specifically designed as an additional way to build on that by utilising the fact that releases are proven in the wild an in large scale.
```
Comment (lizokm) [inline docs/platforms/android/releases/index.mdx:29]:
```suggestion
No. Stability and performance are the top priorities for every release. The `Stable` release status means that the release has been tested and proven to work in the wild and at scale. 
```

File: docs/platforms/android/releases/index.mdx [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+
+### How often do we mark SDKs as stable?
+
+The target cadence is to provide monthly stable releases.
```
Comment (lizokm) [inline docs/platforms/android/releases/index.mdx:33]:
```suggestion
The target cadence is to provide stable releases every month.
```

File: docs/platforms/android/releases/index.mdx [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+
+### Should every customer only use stable releases?
+
+Every developer, team and company knows best which release channel works for them. As stated earlier, every release is considered stable, but `Stable` releases provide an extra layer of safety to make sure the SDK is working as expected, with the tradeoff of a slightly delayed availability. This is also the reason why it's there isn't a pre-release for every release.
```
Comment (lizokm) [inline docs/platforms/android/releases/index.mdx:37]:
```suggestion
### Should I only use stable releases?

This is a personal choice for you, your team, and your company. As stated earlier, every release is considered stable, but `Stable` releases have been tested longer with the tradeoff of a slightly delayed availability. This is also the reason why there isn't a pre-release for every release.
```

File: docs/platforms/android/releases/index.mdx [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+
+### How does this relate to the release process and stability of a feature?
+
+This process is independent of stability of an individual feature and does not change the way features are rolled out. For example, a stable SDK release might still include a new feature that is marked experimental.
```
Comment (lizokm) [inline docs/platforms/android/releases/index.mdx:41]:
```suggestion
This process doesn't affect feature stability and doesn't change the way features are rolled out. For example, a stable SDK release might include a new feature that's marked experimental.
```

**Review by lizokm — APPROVED** [review]
I may be missing something, but it seems to me that instead of saying "release channels" it makes more sense to say "release statuses". Also made a few other wording suggestions. Happy to chat for a few minutes if this doesn't make sense!

File: docs/platforms/android/releases/index.mdx [hunk 11]
```diff
@@ -0,0 +1,41 @@
+---
+title: Release Channels
```
[the remark this answers is not in the document]
Comment (buenaflor) [reply docs/platforms/android/releases/index.mdx:2]: We've adapted the naming of how Google names [their Dart/Flutter releases](https://github.com/flutter/flutter/blob/master/docs/releases/Flutter-build-release-channels.md)

I'm not opposed to changing it to but it's already an established naming (at least in one part of the mobile world) so something to consider as well

[the remark this answers is not in the document]
Comment (brustolin) [reply docs/platforms/android/releases/index.mdx:2]: I think "Statuses" does not convey the actual meaning of what is being described

[the remark this answers is not in the document]
Comment (kahest) [reply docs/platforms/android/releases/index.mdx:2]: @lizokm thanks for the call-out! I agree that we need to make sure it's not mistaken for the Releases/Release Health product. As @buenaflor and @brustolin point out, Release Channels is a somewhat established term for what we're describing here which is why I chose it. Wdyt about changing it to `SDK Release Channels`?

File: docs/platforms/android/releases/index.mdx [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+description: "Learn about the SDK's release channels and choose the right one."
```
[the remark this answers is not in the document]
Comment (kahest) [reply docs/platforms/android/releases/index.mdx:3]: Since this change is only applicable to (specific, i.e. the "big") Mobile SDKs I didn't want to make it sound like this will be used across all of Sentry

File: docs/platforms/android/releases/index.mdx [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+
+No. Stability and performance are the top priorities for every release. The `Stable` release status means that the release has been tested and proven to work in the wild and at scale.
+
+### How often do we mark SDKs as stable?
+
+The target cadence is to provide stable releases every month.
+
+### Should I only use stable releases?
+
+This is a personal choice for you, your team, and your company. As stated earlier, every release is considered stable, but `Stable` releases have been tested longer with the tradeoff of a slightly delayed availability. This is also the reason why there isn't a pre-release for every release.
```
Comment (rahulchhabria) [inline docs/platforms/android/releases/index.mdx:37]: i think we need to reword this to give the reader/user some level of confidence that stable releases are dependable.  being a personal choice doesn't tell the customer that stable releases are actually stable. can we state anything around having a more consistent experience, fewer issues, and will get patched first if we see any issues?

File: docs/platforms/android/releases/index.mdx [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+
+On the releases pages of the SDK's GitHub repo. Stable releases will be indicated with the label `(Stable)` in the name of the release. You can also use the search box in the top right to search for the newest releases.
+
+### Does this mean the release channels are not stable?
+
+No. Stability and performance are the top priorities for every release. The `Stable` release status means that the release has been tested and proven to work in the wild and at scale.
+
+### How often do we mark SDKs as stable?
+
+The target cadence is to provide stable releases every month.
```
Comment (rahulchhabria) [inline docs/platforms/android/releases/index.mdx:33]: when every month?

File: docs/platforms/android/releases/index.mdx [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+
+Pre-releases (alpha, beta, RC,…) are for larger and potentially more impactful changes, such as new features or major versions.
+
+## `Latest`
+
+We continuously release major/minor/hotfix versions from the `main` branch. These releases go through all our internal quality gates and are very safe to use. They're intended to be the default for most teams.
+
+## `Stable`
+
+Releases are promoted from `Latest` to `Stable` when they've been used in the field for some time and at some scale. The length of time a release has been out, the level of adoption, and other quality and stability metrics are considered. These releases will be indicated on the releases page (https://github.com/getsentry/sentry-cocoa/releases/) with the `Stable` label.
```
Comment (philipphofmann) [inline docs/platforms/android/releases/index.mdx:19]:
```suggestion
Releases are promoted from `Latest` to `Stable` when they've been used in the field for some time and at some scale. The length of time a release has been out, the level of adoption, and other quality and stability metrics are considered. These releases will be indicated on the releases page (https://github.com/getsentry/sentry-java/releases/) with the `Stable` label.
```

File: docs/platforms/android/releases/index.mdx (hunk 14 above)
Comment (philipphofmann) [inline docs/platforms/android/releases/index.mdx:33]: Maybe we shouldn't even mention how often we mark them as stable. We could have situations where we mark a release as stable every week and perhaps only every two months.

**Review by philipphofmann — COMMENTED** [review]
Do we plan on adding this to every mobile SDK?

File: docs/platforms/android/releases/index.mdx (hunk 14 above)
Comment (kahest) [reply docs/platforms/android/releases/index.mdx:33]: resolving this convo in favor of https://github.com/getsentry/sentry-docs/pull/11506#discussion_r1791832673

File: docs/platforms/android/releases/index.mdx (hunk 11 above)
[the remark this answers is not in the document]
Comment (kahest) [reply docs/platforms/android/releases/index.mdx:2]: @philipphofmann nope, only specific Mobile SDKs. that said, ppl are viewing this page as part of these Mobile SDK platform's docs, so feels like that would be a bit verbose, and it might be too long for the menu. but no strong opinion against it

File: docs/platforms/android/releases/index.mdx (hunk 13 above)
Comment (kahest) [reply docs/platforms/android/releases/index.mdx:37]: @rahulchhabria wdyt about "As stated earlier, every release is considered stable. On top of that, `Stable` releases have been used in the field for some time and at significant scale, and are the best choice for added dependability and a consistent experience. The tradeoff is a slightly delayed availability."?

For context about "will get patched first": we don't to commit to `Stable` meaning Long Term Support atm, but we will fix recent `Stable` (e.g. current major/minor version, latest `Stable` on previous major) releases on a best effort basis

File: docs/platforms/android/releases/index.mdx (hunk 14 above)
Comment (kahest) [reply docs/platforms/android/releases/index.mdx:33]: no fixed date. we "promote" existing releases that have been out for a while when it makes sense, and we plan to do this at least once a month, but it can depend on release schedule (e.g. prior to a new major there might be fewer stables for the current major)

Comment (kahest) [reply docs/platforms/android/releases/index.mdx:33]: see https://github.com/getsentry/sentry-docs/pull/11506#discussion_r1792180999
I believe it's good to give a rough indication of what the plan is

Comment (kahest) [reply docs/platforms/android/releases/index.mdx:33]: > What about Mobile SDK Release Channels? Or are the other SDKs also applying this process?

see https://github.com/getsentry/sentry-docs/pull/11506#discussion_r1792158408

File: docs/platforms/android/releases/index.mdx (hunk 11 above)
[the remark this answers is not in the document]
Comment (buenaflor) [reply docs/platforms/android/releases/index.mdx:2]: imo at least a search engine result should reflect that it's mobile only

e.g when a user searches for sentry sdk release channels it should show `Mobile SDK Release Channels` instead of only `SDK Release Channels` 

but not sure where this change would need to be added to affect those results

File: docs/platforms/android/releases/index.mdx [hunk 16]
```diff
@@ -0,0 +1,41 @@
...
+
+The target cadence is to provide stable releases every month.
+
+### Should I only use stable releases?
+
+This is a personal choice for you, your team, and your company. As stated earlier, every release is considered stable, but `Stable` releases have been tested longer with the tradeoff of a slightly delayed availability. This is also the reason why there isn't a pre-release for every release.
+
+### How does this relate to the release process and stability of a feature?
+
+This process doesn't affect feature stability and doesn't change the way features are rolled out. For example, a stable SDK release might include a new feature that's marked experimental.
```
[the remark this answers is not in the document]
Comment (kahest) [reply docs/platforms/android/releases/index.mdx:41]: good point, we could make it more explicit. maybe something along the lines of

```This process doesn't affect the release process and stability of specific SDK or Product features, such as Session Replay. For example, a new feature that is still experimental might be part of a stable SDK release, and it will be opt-in and clearly marked as experimental.```

File: docs/platforms/android/releases/index.mdx (hunk 14 above)
Comment (philipphofmann) [reply docs/platforms/android/releases/index.mdx:33]: Maybe we should weaken the sentence by changing it to 
`The target cadence is to provide stable releases roughly every month.`

I'm afraid people will expect a stable release every month and start complaining when there is none. I'm more a fan of it's ready when it's ready. I'm still for removing that part.

File: docs/platforms/android/releases/index.mdx (hunk 13 above)
Comment (rahulchhabria) [reply docs/platforms/android/releases/index.mdx:37]: how about 
> `Stable` releases have been tested in the field and at significant scale. They are intended for a dependable and consistent experience. Please note stable releases will get our newest features later than other releases.

File: docs/platforms/android/releases/index.mdx (hunk 11 above)
[the remark this answers is not in the document]
Comment (kahest) [reply docs/platforms/android/releases/index.mdx]: renamed to `Mobile SDK Releases` - wdyt?

File: docs/platforms/android/releases/index.mdx (hunk 14 above)
Comment (kahest) [reply docs/platforms/android/releases/index.mdx]: reworded to `The stable release cycle depends on a number of factors and can vary, but the target cadence is to provide stable releases roughly every month.` to manage expectations. IMO it's valuable to have this here. if it creates confusion, we'll iterate on the wording (or the process)

File: docs/platforms/android/releases/index.mdx (hunk 13 above)
Comment (kahest) [reply docs/platforms/android/releases/index.mdx]: adopted with minor wording change

File: docs/platforms/android/releases/index.mdx [hunk 17]
```diff
@@ -0,0 +1,41 @@
...
+
+The stable release cycle depends on a number of factors and can vary, but the target cadence is to provide stable releases roughly every month.
+
+### Should I only use stable releases?
+
+Stable releases have been tested in the field and at significant scale. They are intended for a dependable and consistent experience. Please note that the `Stable` release channel will get our newest features later than other releases.
+
+### How does this relate to the stability of a feature?
+
+The `Stable` release status doesn't ascertain the stability of specific SDK or Product features, such as Session Replay. For example, a new feature that is still experimental might be part of a stable SDK release, and it will be opt-in and clearly marked as experimental.
```
Comment (lizokm) [inline docs/platforms/android/releases/index.mdx:41]:
```suggestion
### How does this relate to the stability of a feature?

A `Stable` release status doesn't guarantee the stability of an SDK or product feature. A new feature that's still experimental might be part of a stable SDK release. Any experimental features will be clearly marked and offered as an opt-in.
```
I want to make sure I'm understanding this correctly - are we saying that stable releases won't always have all Sentry features? Or are we saying that we don't guarantee that all features will be stable?

I'm assuming it's that we don't guarantee that all features will be stable b/c some will be experimental. 

If we're saying that we don't guarantee access to all features, we should say: A `Stable` release status doesn't guarantee the stability of a specific SDK or access to all product features.

**Review by lizokm — APPROVED** [review]
Looks great! Just a few more small tweaks.

File: docs/platforms/android/releases/index.mdx (hunk 17 above)
Comment (kahest) [reply docs/platforms/android/releases/index.mdx:41]: > I'm assuming it's that we don't guarantee that all features will be stable b/c some will be experimental.

Yes, that's what we're trying to convey here 👍

File: docs/platforms/android/releases/index.mdx [hunk 18]
```diff
@@ -0,0 +1,41 @@
...
+
+On the <Link to="https://github.com/getsentry/sentry-java/releases/">releases page</Link> of the SDK's GitHub repo. Stable releases will have the `Stable` label in the name of the release. You can also use the search box in the top right to search for the newest releases.
+
+### Does this mean the release channels are not stable?
+
+No. Stability and performance are the top priorities for every release. The `Stable` release status means that the release has been tested and proven to work in the wild and at scale.
+
+### How often do we mark SDKs as stable?
+
+The stable release cycle depends on a number of factors and can vary, but the target cadence is to provide stable releases roughly every month.
```
Comment (philipphofmann) [inline docs/platforms/android/releases/index.mdx:33]: 💯 that's better, thanks.

**Review by philipphofmann — APPROVED** [review]
LGTM, thanks.
